presigned-post.go

```diff
@@ -0,0 +1,224 @@
+package presignedpost
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"time"
+)
+
+// Credentials represent AWS credentials and config.
+type Credentials struct {
+	Region          string // AWS region.
+	Bucket          string // AWS S3 bucket.
+	AccessKeyID     string // AWS access key.
+	SecretAccessKey string // AWS secret access key.
+}
+
+// GetCredentials from the environment.
+func GetCredentials() (*Credentials, error) {
+	// Get AWS credentials from environment variables.
+	region := os.Getenv("AWS_REGION")
+	bucket := os.Getenv("AWS_S3_USER_ASSET_BUCKET")
+	accessKeyID := os.Getenv("AWS_ACCESS_KEY_ID")
+	secretAccessKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
+
+	// Check for missing required environment variables.
+	if region == "" || bucket == "" || accessKeyID == "" || secretAccessKey == "" {
+		return nil, fmt.Errorf("missing required environment variables")
+	}
+
+	// Return the credentials.
+	return &Credentials{
+		Region:          region,
+		Bucket:          bucket,
+		AccessKeyID:     accessKeyID,
+		SecretAccessKey: secretAccessKey,
+	}, nil
+}
+
+// PolicyOptions represent policy options.
+type PolicyOptions struct {
+	ExpiryInMinutes    int // Expiration time in minutes for the policy.
+	MaxFileSizeInBytes int // Maximum allowed file size in the policy.
+}
+
+// PolicyConditions represent policy conditions.
+type PolicyConditions struct {
+	Acl          string // AWS S3 ACL (Access Control List).
+	CacheControl string // Cache control header.
+}
+
+// PolicyTemplate is the policy template.
+const PolicyTemplate = `
+{ "expiration": "%s",
+  "conditions": [
+    {"bucket": "%s"},
+    ["starts-with", "$key", "%s"],
+    {"acl": "%s"},
+    ["content-length-range", 1, %d],
+    {"x-amz-credential": "%s"},
+    {"x-amz-algorithm": "%s"},
+    {"x-amz-date": "%s" },
+    {"Content-Type": "%s"},
+    {"Cache-Control": "%s"},
+  ]
+}
+`
+
+// Policy represents a new policy for uploading sounds.
+type Policy struct {
+	Expiration   string         // Expiration time of the policy.
+	Region       string         // AWS region.
+	Bucket       string         // AWS S3 bucket.
```

Comment:
Bucket is present in Credential struct as well

Reply:
Yes it is required at both the places.